app/jobs/create_story_card_job.rb

```diff
@@ -0,0 +1,98 @@
+class CreateStoryCardJob < ApplicationJob
+  queue_as :default
+  limits_concurrency to: 1, key: ->(story) { story.short_id }, duration: 5.minutes
+
+  CACHE_DIR = Rails.public_path.join("story_image/").to_s.freeze
+
+  def perform(story)
+    return if story.url.blank?
+
+    card_image_url = find_card_image_url(story.url)
+    return if card_image_url.blank?
+
+    story_card_image = generated_card_image(card_image_url)
+
+    if story_card_image.present?
+      FileUtils.mkdir_p(CACHE_DIR)
+
+      File.open("#{CACHE_DIR}/.#{File.basename(story.short_id)}.png", "wb+") do |f|
+        f.write story_card_image
+      end
+
+      File.rename("#{CACHE_DIR}/.#{File.basename(story.short_id)}.png", "#{CACHE_DIR}/#{File.basename(story.short_id)}.png")
+    end
+  end
+
+  private
+
+  def find_card_image_url(url)
+    response = fetch_url(url)
+    return nil unless response
+
+    # If it's a PDF or too large, skip
+    return nil if response.body.length >= 5.megabytes
+    return nil if response["content-type"].to_s.match?(/pdf/)
+
+    converted = response.body.force_encoding("utf-8")
+    parsed = Nokogiri::HTML(converted.to_s)
+
+    # Try <meta property="og:image">
+    card_image_url = parsed.at_css("meta[property='og:image']")&.attributes&.[]("content")&.text
+
+    # Then try <meta name="image">
```

Comment:
I appreciate that you thought to check a few places, the `&.` for handling the complexities of often-invalid pages, and the prudent `rescue` at the end. Good design!